src/input_handler.cc

```diff
@@ -125,13 +125,21 @@ struct MouseHandler
             return true;
 
         case Key::Modifiers::MouseWheelDown:
-            m_dragging = false;
             scroll_window(context, 3);
+            if (m_dragging) {
```

Comment:
I am not sure how that would give correct results as  the `buffer_coord` call is using an out of date, pre-scrolling display buffer. Are you confident you get the correct cursor position on scroll ?

Reply:
If I hold down the button 1 and scroll by one "notch", carefully keeping my pointing device steady, then no: the cursor stays where it was before the scroll occurred, it's not updated to the post-scroll position.

However, after scrolling by a second "notch", the cursor moves to where it was after the first scroll (before the second scroll), so the lag between scrolling and (text) cursor movement is limited. Additionally, in practice after scrolling I move the mouse to the place I where want to end the selection, which updates the cursor position correctly.

If there's a more correct way to scroll the window, though, I'm happy to make changes.

Reply:
It would be a bigger change to get that to work correctly,, however I suspect we can get the exact same behaviour by just removing the `m_dragging = true;` lines. Could you check that ? If thats the case, such a change will be good enough for now.

Reply:
Hmm... do you mean removing all the `m_dragging` handling in `MouseHandler`, including setting `m_dragging` on mouse-press, clearing it on mouse-release and testing it in the mouse-wheel handlers?

If I do that, drag-to-select works just as it did before (since ncurses doesn't send mouse-move events unless a button is held anyway), but now the mouse-wheel always acts like a click as well as a scroll, disrupting the current selection.

Reply:
As discussed on IRC, I tried going back to the original code and just removing the `m_dragging = false;` lines from the mouse-wheel handling.

After that change, scrolling while dragging moves the anchor to the cursor position before the scroll occurs, but it returns to its original position on mouse-move or mouse-release. Scrolling while *not* dragging moves the anchor to the cursor position and it never goes back.